src/System.Private.CoreLib/shared/System/Reflection/StrongNameKeyPair.cs

```diff
@@ -2,57 +2,141 @@
-            byte[] keyPairArray = new byte[length];
-            keyPairFile.Read(keyPairArray, 0, length);
+            var num = (int) keyPairFile.Length;
+            _keyPairArray = new byte[num];
+            keyPairFile.Read (_keyPairArray, 0, num);
+            _keyPairExported = true;
         }
 
         // Build key pair from byte array in memory.
         public StrongNameKeyPair(byte[] keyPairArray)
         {
-            if (keyPairArray == null)
-                throw new ArgumentNullException(nameof(keyPairArray));
+            if (keyPairArray == null) {
+                throw new ArgumentNullException ("keyPairArray");
+            }
+
+            _keyPairArray = new byte[keyPairArray.Length];
+            Array.Copy (keyPairArray, _keyPairArray, keyPairArray.Length);
+            _keyPairExported = true;
         }
 
         protected StrongNameKeyPair(SerializationInfo info, StreamingContext context)
         {
-            throw new PlatformNotSupportedException();
+            _keyPairExported = (bool) info.GetValue ("_keyPairExported", typeof(bool));
+            _keyPairArray = (byte []) info.GetValue ("_keyPairArray", typeof(byte []));
+            _keyPairContainer = (string) info.GetValue ("_keyPairContainer", typeof(string));
+            _publicKey = (byte []) info.GetValue ("_publicKey", typeof(byte []));
         }
 
         public StrongNameKeyPair(string keyPairContainer)
         {
-            throw new PlatformNotSupportedException(SR.PlatformNotSupported_StrongNameSigning);
+            if (keyPairContainer == null) {
+                throw new ArgumentNullException ("keyPairContainer");
+            }
+
+            _keyPairContainer = keyPairContainer;
+            _keyPairExported = false;
         }
 
         public byte[] PublicKey
         {
-            get
-            {
-                throw new PlatformNotSupportedException(SR.PlatformNotSupported_StrongNameSigning);
+            [SecuritySafeCritical]
+            get {
+                if (_publicKey == null) {
+                    _publicKey = ComputePublicKey ();
+                }
+
+                byte [] array = new byte[_publicKey.Length];
+                Array.Copy (_publicKey, array, _publicKey.Length);
+                return array;
             }
         }
 
         void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
         {
-            throw new PlatformNotSupportedException();
+            info.AddValue ("_keyPairExported", _keyPairExported);
+            info.AddValue ("_keyPairArray", _keyPairArray);
+            info.AddValue ("_keyPairContainer", _keyPairContainer);
+            info.AddValue ("_publicKey", _publicKey);
         }
 
         void IDeserializationCallback.OnDeserialization(object sender)
         {
-            throw new PlatformNotSupportedException();
+        }
+
+        byte [] ComputePublicKey ()
+        {
+            using (var rsa = Mixin.CreateRSA (this)) {
```

Comment:
Crypto is not available in CoreLib. This would need to be all done via reflection.